File: Modules/SMES/Tasks/SIE.Web.MES.TaskManagement/Dispatchs/Commands/PauseTaskCommand.cs
using SIE.MES.TaskManagement.Dispatchs;
using SIE.Web.Command;
using System;
using System.Linq;

namespace SIE.Web.MES.TaskManagement.Dispatchs.Commands
{
    /// <summary>
    /// 暂停任务单
    /// </summary>
    [JsCommand("SIE.Web.MES.TaskManagement.Dispatchs.Commands.PauseTaskCommand")]
    public class PauseTaskCommand : ViewCommand
    {
        /// <summary>
        /// 执行暂停操作
        /// </summary>
        /// <param name="args">args</param>
        /// <param name="scope">scope</param>
        /// <returns>暂停结果</returns>
        protected override object Excute(ViewArgs args, string scope)
        {
            string errMsg = string.Empty;
            var selectedIds = args.SelectedIds.ToList();
            errMsg = RT.Service.Resolve<DispatchController>().SetPauseTasks(selectedIds);
            if (errMsg.Length == 0)
                return "暂停成功";
            else
                return errMsg;
        }
    }
}

File: Modules/SMES/Tasks/SIE.Web.MES.TaskManagement/Dispatchs/Commands/PauseTaskCommand.js
SIE.defineCommand('SIE.Web.MES.TaskManagement.Dispatchs.Commands.PauseTaskCommand', {
    meta: { text: "暂停", group: "edit", hierarchy: "状态", iconCls: "icon-Pause icon-blue" },
    canExecute: function (view) {
        var selecteditems = view.getSelection();
        if (selecteditems != null && selecteditems.length > 0) {
            for (var i = 0; i < selecteditems.length; i++) {
                if (selecteditems[i].data.TaskStatus === 40 || selecteditems[i].data.TaskStatus === 50 || selecteditems[i].data.TaskStatus === 60) {
                    return false;
                }
            }
            return true;
        }
        return false;
    },
    execute: function (view, source) {
        view.execute({
            withIds: true,
            selectIds: view.getSelectionIds(),
            success: function (res) { //回调
                var errMsg = res.Result;
                if (errMsg == '暂停成功')
                    view.reloadData();
                SIE.Msg.showMessage(errMsg.t());
            }
        });
    },
});